models/zeroex/polygon/erc20/zeroex_api_polygon_fills.sql

```diff
@@ -0,0 +1,390 @@
+    SELECT tx_hash,
+           max(affiliate_address) as affiliate_address, temp.to, temp.from, temp.block_number
+    FROM (
+        SELECT tr.tx_hash,
+                    '0x' || CASE
+                                WHEN POSITION('869584cd' IN INPUT) <> 0
+                                THEN SUBSTRING(INPUT
+                                        FROM (position('869584cd' IN INPUT) + 32)
+                                        FOR 40)
+                                WHEN POSITION('fbc019a7' IN INPUT) <> 0
+                                THEN SUBSTRING(INPUT
+                                        FROM (position('fbc019a7' IN INPUT) + 32)
+                                        FOR 40)
+                            END AS affiliate_address, tr.to as to, 
+                            tr.from as from, tr.block_number as block_number
+        FROM {{ source('polygon', 'traces') }} tr
+        WHERE tr.to IN (
+                -- exchange contract
+                '0x61935cbdd02287b511119ddb11aeb42f1593b7ef', 
+                -- forwarder addresses
+                '0x6958f5e95332d93d21af0d7b9ca85b8212fee0a5',
+                '0x4aa817c6f383c8e8ae77301d18ce48efb16fd2be',
+                '0x4ef40d1bf0983899892946830abf99eca2dbc5ce', 
+                -- exchange proxy
+                '0xdef1c0ded9bec7f1a1670819833240f027b25eff'
+                )
+                AND (
+                        POSITION('869584cd' IN INPUT) <> 0
+                        OR POSITION('fbc019a7' IN INPUT) <> 0
+                    )
+                
+                {% if is_incremental() %}
+                AND block_time >= date_trunc('day', now() - interval '1 week') 
+                {% endif %}
+                {% if not is_incremental() %}
+                AND block_time >= '{{zeroex_v3_start_date}}'
+                {% endif %}
+    ) temp
+    group by tx_hash, to, from, block_number
+
+),
+
+v4_rfq_fills_no_bridge AS (
+    SELECT 
+            fills.evt_tx_hash               AS tx_hash,
+            fills.evt_index,
+            fills.contract_address,
+            fills.evt_block_time            AS block_time,
+            fills.maker                     AS maker,
+            fills.taker                     AS taker,
+            fills.takerToken                AS taker_token,
+            fills.makerToken                AS maker_token,
+            fills.takerTokenFilledAmount    AS taker_token_amount_raw,
+            fills.makerTokenFilledAmount    AS maker_token_amount_raw,
+            'RfqOrderFilled'                AS type,
+            zeroex_tx.affiliate_address     AS affiliate_address,
+            zeroex_tx.to, zeroex_tx.from, zeroex_tx.block_number,
+            (zeroex_tx.tx_hash IS NOT NULL) AS swap_flag,
+            FALSE                           AS matcha_limit_order_flag
+    FROM {{ source('zeroex_polygon', 'ExchangeProxy_evt_RfqOrderFilled') }} fills
+    LEFT JOIN zeroex_tx ON zeroex_tx.tx_hash = fills.evt_tx_hash
```

Comment:
@RantumBits this is likely the reason for CI tests timing out & models taking longer to run -- left join vs. inner join from decode tables to base tables. is it feasible to make all of these joins inner to `zeroex_tx`? any reason it has to be left join? if it's feasible, please do push a change on all CTE queries for both polygon and optimism, we can see if timing improves.

Reply:
ty for the help ser...giving that a try now

Reply:
timed out again. not clear why it's no longer completing compared to previous version

Reply:
it could be combination of running both polygon & optimism at same time, plus potentially other unrelated models from other PRs active at same time. let me try to run locally, see what timing looks like

Reply:
if you could hold off on commits for a bit, i'll see what it looks like

Reply:
it's polygon which is taking a long time on initial full historical load. it makes sense if the data volume is large. i'll ensure tests run as expected locally, then we can bypass the timeouts here if all looks good.

Reply:
Yeah OP is not a problem. Only poly causing issues. I was able to get it to complete last week but timing out now

Reply:
i think i'll need to break the query down and analyze the query plan. first glance, without edits, the job was reading 144 billion+ rows. that doesn't seem right.

Reply:
hmm...that doesn't seem right. i'll take another look and compare to the front-end query, which completes fairly quickly

Reply:
may have been a bug on my side. i made minor tweaks locally without pushing yet. i'm close to having some metrics, where i can push changes here and share the numbers. if it's as expected, we can proceed without the CI test passing as i've done so locally.

Reply:
awesome, really appreciate your help with this